Dockerfile

```diff
@@ -70,7 +70,8 @@ RUN set -eux; \
     groupadd -r tox --gid=10000; \
     useradd --no-log-init -r -g tox -m --uid=10000 tox; \
     mkdir /tests; \
-    chown tox:tox /tests
+    chown tox:tox /tests; \
+    chown tox:tox /home/tox
```

Comment:
I guess the problem that you're trying to solve is that if you mount a Docker volume at `/home/tox/.cache` then it's owned by `root`:

```console
$ docker run -it --rm -v tox_cache:/home/tox/.cache 31z4/tox-test-minimal-flask ls -la /home/tox/     
total 28
drwxr-x--- 1 tox  tox  4096 Oct 12 13:13 .
drwxr-xr-x 1 root root 4096 Oct  3 12:51 ..
-rw-r--r-- 1 tox  tox   220 Jan  6  2022 .bash_logout
-rw-r--r-- 1 tox  tox  3771 Jan  6  2022 .bashrc
drwxr-xr-x 2 root root 4096 Oct 12 12:51 .cache
-rw-r--r-- 1 tox  tox   807 Jan  6  2022 .profile
```

Therefore, tox cache couldn't work.

But just changing owner during image build won't help. In fact, `/home/tox` is already owned by `tox:tox`:

```console
$ docker run -it --rm 31z4/tox ls -l /home                                                 
total 4
drwxr-x--- 2 tox tox 4096 Oct  3 12:51 tox
```

However, if you create `/home/tox/.cache` and do `chown tox:tox /home/tox/.cache` the permission issue should be solved. I did a quick test locally (I built `tox` image with changes that I suggested):

```console
$ docker run -it --rm tox ls -la /home/tox/
total 24
drwxr-x--- 3 tox  tox  4096 Oct 12 13:26 .
drwxr-xr-x 1 root root 4096 Oct 12 13:26 ..
-rw-r--r-- 1 tox  tox   220 Jan  6  2022 .bash_logout
-rw-r--r-- 1 tox  tox  3771 Jan  6  2022 .bashrc
drwxr-xr-x 2 tox  tox  4096 Oct 12 13:26 .cache
-rw-r--r-- 1 tox  tox   807 Jan  6  2022 .profile
```

```console
$ docker run -it --rm -v tox_cache:/home/tox/.cache tox ls -la /home/tox/
total 24
drwxr-x--- 3 tox  tox  4096 Oct 12 13:26 .
drwxr-xr-x 1 root root 4096 Oct 12 13:26 ..
-rw-r--r-- 1 tox  tox   220 Jan  6  2022 .bash_logout
-rw-r--r-- 1 tox  tox  3771 Jan  6  2022 .bashrc
drwxr-xr-x 2 tox  tox  4096 Oct 12 13:26 .cache
-rw-r--r-- 1 tox  tox   807 Jan  6  2022 .profile
```

Reply:
To be more specific here's the changes that I suggest:

```diff
diff --git a/Dockerfile b/Dockerfile
index 6f41e37..82a0a4f 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -69,8 +69,8 @@ RUN set -eux; \
     pip3.11 install --no-deps -r /requirements.txt; \
     groupadd -r tox --gid=10000; \
     useradd --no-log-init -r -g tox -m --uid=10000 tox; \
-    mkdir /tests; \
-    chown tox:tox /tests
+    mkdir /tests /home/tox/.cache; \
+    chown tox:tox /tests /home/tox/.cache
 
 WORKDIR /tests
 VOLUME /tests
```

Reply:
You are right! I must have missed that for whatever reason.